documentation/matlab-api.md

```diff
@@ -0,0 +1,385 @@
+```matlab
+function_name(a, b, c, 'flagname')
+function_name(a, b, c, '-flagname')
+function_name(a, b, c, '-noflagname')
+function_name(a, b, c, '-f')
+function_name(a, b, c, '-nof')
+```
+This flexibility allows less typing, but at the same time may obscure the intended operation or even lead to the wrong flag being used.
+
+
+#### Python
+
+```python
+def function_name(a, b, c, flagname = true) #default
+
+def function_name(a, b, *args, **kwargs):
+    if any(arg.startswith('-') for arg in args): # Handle flags
+        pass
+    # args = [value-of-c, ...]
+    # kwargs = {filename: true, ...}
+```
+As Python does not allow args starting with `-`, this will require some way of handling the flag.
+```python
+def function_name(a, -flag) # ERROR
+```
+
+This can either be by passing flags through as a string pre-parsing the input before the call to MATLAB to be handled by `*args`
+```python
+def function_name(a, b, *args, **kwargs)
+    if any(arg.startswith('-') for arg in args if isinstance(arg, string)): # Handle flags
+       pass
+```
+Or passed directly through as one of args.
+```python
+def function_name(a, b, *args, **kwargs):
+    call_MATLAB(a, b, *args, **kwargs)
+```
+
+Which would be called as:
+```python
+function_name(a, b, '-flag')
+```
+
+Or by passing in through dict form to be handled by `**kwargs` (nobody would do this).
+```python
+function_name(a, b, **{'-flag':True})
+```
+
+This will depend on the ultimate interface `call_MATLAB` has.
+
+#### Pros and Cons
+
+##### Pros
+
+- Less typing for the average user?
+```matlab
+function_name(a, '-flag')
+function_name(a, 'flag', 1)
+function_name(a, '-noflag')
+function_name(a, 'flag', 0)
+```
+- Python can just pass args directly into the MATLAB call in principle
+
+##### Cons
+
+- Raises questions about the separation between arguments and flags are additional arguments to be labelled as flags?
+- Not Pythonic way of doing things
+- Requires special parsing of `-`'d args using own parser
+- MATLAB seems to be moving towards keyword-value scheme
+- Potentially inconsistent, ambiguous or multiplicitous arguments (in terms of `-disable-x` vs `-nodisable-x` vs `-enable-x` vs `-noenable-x` vs `-x`)
+- Requires mixed syntax for passing keyword values anyway
+
+#### Transition implications
+
+- Requires potential workaround in Python
+- Requires developing tests for `parse_arguments`
+- Requires keeping `parse_arguments` up to date and functional
+
+### Keyword args
+#### MATLAB
+
+`inputParser` or PACE's `parse_arguments` would work here.
+
+MATLAB keyword arguments are commonly passed through as:
+
+```matlab
+function_name(a, b, c, 'flagname', true)
+function_name(a, b, c, 'flagname', false)
+```
+
+#### Python
+
+Python keyword arguments are commonly passed through as:
+
+```python
+function_name(a, b, c, flagname = true)
+function_name(a, b, c, flagname = false)
+```
+
+If `call_MATLAB` is implemented along the lines of:
```

Comment:
This is exactly how it is currently implemented: https://github.com/pace-neutrons/pace-python/blob/main/pace_python/MatlabProxyObject.py#L104